test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -328,6 +328,30 @@ def forward(self, x, y):
                     mod, (v, other), match_count, match_nodes, rtol=1e-2, atol=1e-2
                 )
 
+    def test_multi_linear_share_same_input(self):
+        # llama pattern.
+        class M(torch.nn.Module):
+            def __init__(
+                self,
+            ):
+                super().__init__()
+                self.w1 = torch.nn.Linear(16, 16, bias=False)
+                self.w2 = torch.nn.Linear(16, 16, bias=False)
+
+            def forward(self, x):
+                return F.silu(self.w1(x)) * self.w2(x)
```

Comment:
if w2 also has a post-op, would the fusion happen too?

Reply:
yes

Reply:
Do we have UT to cover that too? Please add it if not, otherwise, I'm fine with the PR.

Reply:
Now, update this test case to cover that.